src/types/index.d.ts

```diff
@@ -562,7 +562,9 @@ export interface ZoneEgressOverview extends MeshEntity {
     zone?: string
     networking?: ZoneEgressNetworking
   }
-  zoneEgressInsight: any
+  zoneEgressInsight?: {
+    subscriptions: DiscoverySubscription[]
+  }
```

Comment:
**Issue**: I don’t thin k`zoneEgressInsight` can be omitted, but `subscriptions` can.

```suggestion
  zoneEgressInsight: {
    subscriptions?: DiscoverySubscription[]
  }
```

Reply:
Well I'm not totally sure these are the generated types:

![Screenshot 2023-09-28 at 10 08 38](https://github.com/kumahq/kuma-gui/assets/554604/b400cef8-8030-4aa9-b7cd-89b27b6299cb)
![Screenshot 2023-09-28 at 10 10 19](https://github.com/kumahq/kuma-gui/assets/554604/85acd364-efcf-43d0-865c-5f6cbd3b42ad)

?

Reply:
but then again this is DataplaneInsights, which we know, I guess without the extra protections that the generated types give, can have `subscriptions` as `undefined`

![Screenshot 2023-09-28 at 10 14 52](https://github.com/kumahq/kuma-gui/assets/554604/8aae9252-a083-43a2-bd1f-deaee9ca866a)

This is were I felt I was going down a large rabbit hole, so I'm more inlined to remove this change entirely. Hopefully we'll be sorting this whole issue out once and for all soonish.

Give me a bit let me take a better look at this from here. brb

Reply:
Well the point is, we shouldn't have to keep that in mind as these generated types make sure that subscriptions is always an array, which is what we want.

![Screenshot 2023-09-28 at 13 18 41](https://github.com/kumahq/kuma-gui/assets/554604/5e20df78-cb8b-4b1b-9b08-f679ef53665d)

>  It definitely looks there is a knowledge gap somewhere we because we do indeed know subscriptions can be undefined.

Well, I'm not sure if its a "knowledge gap" thing. With those generated types `subscriptions` is always an array, which then prevents all the undefined access errors we have without having to keep that in mind or having to have the knowledge about it. I think that's part of the point.

1 sec I think I'm going to take these type changes out as I'm guessing they could have a bigger impact that I don't want to have to look at in this PR and this was part of the rabbit hole I got the feeling I was getting into here.

Reply:
OK I removed the type changes here, I have an approval already so I'll wait a bit before merging just incase.